lib/templates.ts

```diff
@@ -1145,61 +1145,26 @@ export async function updateResponseDeliveryOption(
  * @param formID The unique identifier of the form you want to modify
  * @returns The updated form template or null if the record does not exist
  */
-export async function removeDeliveryOption(
-  ability: UserAbility,
-  formID: string
-): Promise<FormRecord | null> {
+export async function removeDeliveryOption(ability: UserAbility, formID: string): Promise<void> {
   try {
     await authorization.canEditForm(ability, formID);
 
-    const updatedTemplate = await prisma.template
-      .update({
+    await prisma.deliveryOption
```

Comment:
Just confirming that we're allowing the `removeDeliveryOption` to run on published forms? 
The initial reason to use the `prisma.template.update()` was the ability to check if the form was published in the unique `where:{ }` before deleting the delivery option.

Reply:
I'll take this back to draft, as there seems to be another bug introduced.

But yeah maybe this will just have to be two queries either way:

Check to make sure it's not published, then delete.
OR
Check to see if there is a DeliveryOption and use the update syntax.

The update syntax fails if there was no DeliveryOption set.

Reply:
We can also catch that specific error in the prisma chain and ignore it.

Reply:
Seems to throw the same error in both cases.

Template with a DeliveryOption and published = true : P2025

And

Template with *no* DeliveryOption and published = false : P2025 

...doesn't make a whole lot of sense as it says "No 'DeliveryOption' record was found ..." in both cases.

Reply:
Pushed a commit with the split check, then delete.
I couldn't get it to work with just the update call.

Reply:
If we want it to fail silently do we even need to return a descriptive error if the template is already published?

Reply:
In other words shouldn't the client side be verifying if the template is already published before trying to make the action call?  And if someone tries to hit the action endpoint directly it just returns a generic error.

Reply:
For the "live api key version" --- there is code that does the published check.

... depends on where we calling the code from.

Part of this was looking to ensure the server is cleaned up when saving to "the vault".

Reply:
that said looks like Dave already made some further edits and checks